### DISCUSSION TITLE
Why is the old package deprecated before this one is feature-complete?

### DISCUSSION THREAD

**Discussion by buzzb0x** [LottieFiles/dotlottie-web#154] [General]
Hello,

Thanks for the development of this new and improved library. I noticed you added a deprecation notice to the old one last week, and we started using Lottie on a project I'm working on so we switched to this one.

We're now needed to use multiple animations in the same dotlottie as well as state machines, and this package doesn't support any of that. Furthermore, no documentation is up to date on the website yet.

Should we just switch back to the old package? Maybe be clearer on the documentation and on the old package that this is an early release and only supports the most basic use cases?

Cheers

**Comment by theashraf** [comment]
Thanks, @buzzb0x, for your feedback.

To address the issue of inconsistent Lottie rendering experiences across platforms (Mobile and Web), we're developing a cross-platform, performant [player in Rust](https://github.com/LottieFiles/dotlottie-rs). This development has led us to focus on the dotlottie-web, along with the [dotlottie-android](https://github.com/LottieFiles/dotlottie-android) and [dotlottie-ios](https://github.com/LottieFiles/dotlottie-ios) players, which are built on top of [dotlottie-rs](https://github.com/LottieFiles/dotlottie-rs). As part of this shift, we've decided to deprecate the old package @dolottie/player-component, recommending dotlottie-web as its successor.

I understand there's currently a lack of documentation for the new dotLottie-web player. I assure you that I will ensure it's updated soon and will keep you informed on the progress. In the meantime, for use cases and usage, you can refer to the READMEs of dotlottie-web and the [CodePen examples](https://codepen.io/collection/rxPaBN).

The dotlottie-web player supports almost all the features of the old package, including the ability to play .lottie multi-animations. We are still working on adding state machine support, which we expect to complete later this year. Although the `@dotlottie/player-component` package is now deprecated, it remains available for use if it suits your current needs. However, please be aware that we have halted any future updates and support for this package.

**Reply by buzzb0x** [reply]
Thanks for your quick answer. My main hope is that you're able to improve this package quickly and add more declarative bindings to the framework wrappers.

**Reply by theashraf** [reply]
@buzzb0x, here are the currently available dotlottie-web framework wrappers:

- [dotlottie-react](https://www.npmjs.com/package/@lottiefiles/dotlottie-react)
- [dotlottie-vue](https://www.npmjs.com/package/@lottiefiles/dotlottie-vue)
- [dotlottie-svelte](https://www.npmjs.com/package/@lottiefiles/dotlottie-svelte)
- [dotlottie-wc](https://www.npmjs.com/package/@lottiefiles/dotlottie-wc)

If you'd like to add another component wrapper, please submit an issue for it, and we will discuss it further.

**Reply by theashraf** [reply]
@buzzb0x, You can refer to these docs for dotlottie-web and its wrappers usage and APIs:

https://developers.lottiefiles.com/docs/dotlottie-web/
